File: loggalitic/src/main/java/com/android/oleksandrpriadko/loggalitic/policy/DefaultPolicy.java
package com.android.oleksandrpriadko.loggalitic.policy;

import android.text.TextUtils;
import android.util.Log;

import com.android.oleksandrpriadko.loggalitic.analytics.AnalyticsEvent;


/**
 * Created by Oleksandr Priadko.
 * 7/9/17
 */
public class DefaultPolicy extends Policy {

    public DefaultPolicy(int logLevel, int logToEventLevel, boolean debug) {
        super(logLevel, logToEventLevel, debug);
    }

    @Override
    public boolean isEventAllowed(AnalyticsEvent event) {
        if (TextUtils.isEmpty(event.getName())) {
            Log.e(getTag(), "analytics event not allowed: name of event is empty");
            return false;
        }
        return true;
    }

    @Override
    public boolean isLogAllowed(String tag, int level, boolean force) {
        return force || isLogAllowed(tag, level);
    }

    @Override
    public boolean isLogReportAllowed(String tag, int level, boolean force) {
        return level >= this.getLogToEventLevel();
    }

    @Override
    public boolean isLogAllowed(String tag, int level) {
        return level <= this.getLogLevel();
    }
}

File: loggalitic/src/main/java/com/android/oleksandrpriadko/loggalitic/policy/DefaultPolicy.kt
package com.android.oleksandrpriadko.loggalitic.policy

import android.text.TextUtils
import com.android.oleksandrpriadko.loggalitic.LogPublishService
import com.android.oleksandrpriadko.core.policy.AnalyticsEvent
import com.android.oleksandrpriadko.core.policy.Policy

class DefaultPolicy : Policy {

    constructor() : super(ASSERT, ASSERT, false)

    constructor(logLevel: Int, logToEventLevel: Int, debug: Boolean) : super(logLevel, logToEventLevel, debug)

    override fun isEventAllowed(event: AnalyticsEvent): Boolean {
        if (TextUtils.isEmpty(event.name)) {
            LogPublishService.logger().e(tag, "analytics event not allowed: name of event is empty")
            return false
        }
        return true
    }

    override fun isLogAllowed(tag: String, level: Int, force: Boolean): Boolean {
        return force || isLogAllowed(tag, level)
    }

    override fun isLogReportAllowed(tag: String, level: Int, force: Boolean): Boolean {
        return level >= this.logToEventLevel
    }

    override fun isLogAllowed(tag: String, level: Int): Boolean {
        return level <= this.logLevel
    }
}
